### DISCUSSION TITLE
Flux NF4 model has been released

### DISCUSSION THREAD

**Discussion by xingewh** [Acly/krita-ai-diffusion#1076] [General]
Flux NF4 model has been released, and it seems to have better quality and faster speed than FP8.” 或者 “The Flux NF4 model has been launched, and it appears to possess better quality and a higher speed than FP8.

https://huggingface.co/lllyasviel/flux1-dev-bnb-nf4/blob/main/flux1-dev-bnb-nf4.safetensors

**Comment by gmf1982** [comment]
It has been used in comfyui. My graphics card is 2060super 8GB. Using flux1-schnell-nf4.safetensors, it only takes 23 seconds to generate a 1024*1024 picture. The effect is very good. I hope krita-ai-diffusion will support NF4 soon.

**Comment by xingewh** [comment]
V2:
https://huggingface.co/lllyasviel/flux1-dev-bnb-nf4/blob/main/flux1-dev-bnb-nf4-v2.safetensors

**Reply by AlexYez** [reply]
Cool, this version don't need separate clip encoders and vae

**Reply by xingewh** [reply]
> 很酷，这个版本不需要单独的夹子编码器和 vae

Yes, in fact, we only need to install this node https://github.com/comfyanonymous/ComfyUI_bitsandbytes_NF4. 
Then edit the comfy_workflow.py, and replace return self.add_cached("CheckpointLoaderSimple", 3, ckpt_name=checkpoint) with return self.add_cached("CheckpointLoaderNF4", 3, ckpt_name=checkpoint), then we can use the NF4 model in Krita AI.
I have successfully used the NF4 v2 model in my Krita AI.
![屏幕截图 2024-08-15 214401](https://github.com/user-attachments/assets/e6a96e73-7aa6-4b5b-a580-33dd08cef0e5)
![屏幕截图 2024-08-15 214344](https://github.com/user-attachments/assets/92c07a9d-4d76-4d1f-a223-cf8c7237ac0a)

**Reply by xingewh** [reply]
However, after installing this node, some errors occurred. I'm not sure whether their specific reasons are related to it.

**Reply by xingewh** [reply]
After testing, after I uninstalled the ComfyUI_bitsandbytes_NF4 node, the malfunction where some sdxl models generated error prompts when generating pictures was eliminated. It might be that it conflicts with some current components.

**Reply by yen-shi-lun** [reply]
Thank you very much, I will try it too

**Reply by mliistro-beep** [reply]
Hi, I'm a bit late .. I'm not sure but in my kritadiffusion there's no comfy_workflow.py. Is my kritadiffusion installation broke or something changed?
Tnks

**Comment by xingewh** [comment]
![屏幕截图 2024-08-15 193939](https://github.com/user-attachments/assets/ad1e9184-a0d5-47a2-91b1-be4a94fb225a)
I conducted a test. On my 4070Ti 12G, the NF4 version takes only 32 seconds to generate a 1024*1024 image in 20 steps.

**Comment by nesmonmehdydeal** [comment]
In Krita those zip do not intalled in plugin inpot inside Krita 
 - https://github.com/city96/ComfyUI-GGUF
 - https://github.com/comfyanonymous/ComfyUI_bitsandbytes_NF4 
How to do it manually ?

**Reply by xingewh** [reply]
I am using an independent ComfyUI. Nodes can be installed by scanning through plugin management or manually added to the custom_nodes directory for installation.

**Reply by nesmonmehdydeal** [reply]
Thanks.

**Comment by tilllt** [comment]
Is there a simple to follow how-to on installing flux models with krita-ai-diffusion?  I dropped the models into the model folder but the config doesn't seem to "see" them...

**Reply by ibow1983** [reply]
me too！！

**Reply by gmf1982** [reply]
Try updating comfyui-inpaint-nodes and comfyui-tooling-nodes

**Reply by tilllt** [reply]
update as in redownload the nodes and overwrite their old code or is there a specific procedure to update nodes in comfyUI (have rarely used comfUi standalone, just in Krita.

**Reply by gmf1982** [reply]
You can download the node overwrite, or you can go to custom_nodes>node that needs to be updated>enter CMD in the address bar to open the terminal, and update it with the “git pull“” command.

**Reply by yecai2024** [reply]
I updated all the nodes, but Krita still can't find all flux models.
![Q911](https://github.com/user-attachments/assets/920ba454-29c7-431a-b36a-1392d39e3209)


> Try updating comfyui-inpaint-nodes and comfyui-tooling-nodes

**Reply by gmf1982** [reply]
step1:update Comfyui
step2:updating comfyui-inpaint-nodes and comfyui-tooling-nodes(Make sure you have updated successfully)
step3:Download the [model](https://huggingface.co/Comfy-Org/flux1-schnell/resolve/main/flux1-schnell-fp8.safetensors) and put it in the models/checkpoints folder. 
step4:Restart Comfyui and Krita
This is the solution I had when I encountered the same problem。

![PixPin_2024-09-12_13-10-09](https://github.com/user-attachments/assets/82022bf1-7e9d-4a19-90a6-ffdc3be975ee)
![PixPin_2024-09-12_13-08-04](https://github.com/user-attachments/assets/2ed078ca-24b7-47a5-b715-6043a57d48b8)

**Comment by tilllt** [comment]
I tried upgrading the nodes, I even tried reinstalling a standalone ComfyUI now, the results are the same: the images come out extremely "ghostly", blurry, like you can just guess what the prompt was suppsoed to create. Is there something like a step-by-step walkthrough how to install flux, with which models (including fp4/fp8) etc to get it to work?
